docs/trex_tableau_viz_ref.md

```diff
@@ -0,0 +1,215 @@
+
+
+## JSON specification for `inputSpec` object
+
+The `inputSpec` is a JavaScript object that uses JSON format and specifies the attribute-value pairs to use to create the SVG image. The object consists of the `data` and the attributes that specify how to format the data, for example, `mark`, `markcolor`, `size`, and `encoding`. The `data` is an array of objects. The `data` is listed in row-level order, where each object represents one item. The `mark` attribute specifies the type of mark to use in the image,for example, you can change the marks from bars to lines. 
+
+The following shows an example `inputSpec` that creates a bar chart image.
+
+```json
+
+
+{
+  "description": "A bar chart with multiple encodings",
+  "size": {"width": 800, "height": 600},
+  "data": {
+    "values": [
+      {"Category": "A", "Sales": 28, "Weather": "Sun", "Quantity": 3},
+      {"Category": "B", "Sales": 55, "Weather": "Sun", "Quantity": 6},
+      {"Category": "C", "Sales": 43, "Weather": "Sun", "Quantity": 9}
+    ]
+  },
+  "mark": "bar",
+  "markcolor": "#FFA500",
+  "encoding": {
+    "columns": {"field": "Sales", "type": "continuous"},
+    "rows": {"field": "Category", "type": "discrete", "hidden": "true"},
+    "color": {"field": "Weather", "type": "discrete", "palette": "seattle_grays_10_0"},
+    "size": {"field": "Quantity", "type": "continuous"},
+    "text": {"field": "Category", "type": "discrete"}
+  }
+}
+
+
+```
+
+---
+## Descripion of `inputSpec` keys
+
+The following table describes the keys of the elements in the `inputSpec`, the JavaScript JSON object.
+
+
+### `description`
+
+(Optional)    A description about the visualization. Currently, the description is not used anywhere and is not visible when the image is generated. However, it might be helpful to use this description as you comment your code.
+
+### `title`
+
+(Optional)    Creates a title for the image.
+
+```json
+title: "My Custom Viz", 
+
+```
+
+### `data`
+
+(Required) The data to use to create the image. The `data` consists of an array of objects, specified with the `"values"` key. Data is listed in row level order, where each object represents one item.
+
+```json
+data: {
+  values: [
+   { Category: 'A', Sales: 28 },
+   { Category: 'B', Sales: 55 },
+   { Category: 'C', Sales: 43 },
+   { Category: 'D', Sales: 91 },
+   { Category: 'E', Sales: 81 },
+   { Category: 'F', Sales: 53 },
+   { Category: 'G', Sales: 19 },
+   { Category: 'H', Sales: 87 },
+   { Category: 'I', Sales: 52 }
+  ]
+},
+
+```
+
+### `mark`
+
+(Optional) Supported mark types include: `bar`, `line`, `area`, `square`, `circle`, `text`. If no mark type is specified, the generated image will use the Tableau default for that data.
```

Comment:
We have a type that should be used for mark: tableau.MarkType.

Reply:
A couple of questions. 
1. Do we also support the string literals?  From my testing, specifying `'bar'`, `'line'`, etc. works fine. It seems simpler, but then it might be bad style. Should we list both? Or just the `tableau.MarkType` types `tableau.MarkType.Bar`, etc.? 

2.  We don't currently support all the mark types listed for `tableau.MarkType`. It's my understanding that the only types currently supported are the ones listed here: bar, line, area, square, circle, and text.  However, in the Extensions API reference docs, we list all the enumerations for `tableau.MarkType` - should those be labeled as "not implemented" ?  If they don't work, it might confuse people.

Reply:
We can document the types we currently support. MarkType will be shared by another API, so we can't put not implemented in there.

Rather than respond to all comments about enums versus literals I'll editorialize here...
Yes, they can put a literal in there and it will work. For example 'bar' or 'area'. The benefit of the enums are the following:
1. In most cases (except MarkType) it constrains them to valid options only.
2. Autocomplete will happen in many editors
3. It avoids simple spelling errors like 'continous' rather than 'continuous'. 
4. It is best practice. Let's teach best practice in our documentation.

Reply:
! agree 100%. Updating these instances.